src/upgrades/upgrade.py

```diff
@@ -226,7 +226,7 @@ def authorized(self) -> bool:
         """
 
     @abc.abstractmethod
-    def upgrade_unit(self, *, snap) -> None:
+    def upgrade_unit(self, charm) -> None:
```

Comment:
```suggestion
    def upgrade_unit(self, *, charm) -> None:
```
nit: to keep some API consistency with other charms